src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+
+        # enforce differential order constraints
+        if np.any(np.isneginf(self.differential_order)):
+            logger.error(
+                '`APWBaseOrbital.differential_order` must be completely non-negative.'
+            )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # it's hard to enforce commutative diagrams between `_determine_apw` and `normalize`
+        # instead, make all `_determine_apw` soft-coupled and dependent on the normalized state
+        # leverage normalize ∘ normalize = normalize
+        self._is_normalized = False
+
+
+class APWOrbital(APWBaseOrbital):
+    """
+    Implementation of `APWWavefunction` capturing the foundational (S)(L)APW basis sets, all of the form $\sum_{lm} \left[ \sum_o c_{lmo} \frac{\partial}{\partial r}u_l(r, \epsilon_l) \right] Y_lm$.
+    The energy parameter $\epsilon_l$ is always considered fixed during diagonalization, opposed to the original APW formulation.
+    This representation then has to match the plane-wave $k_n$ points within the muffin-tin sphere.
+
+    * D. J. Singh and L. Nordström, \"INTRODUCTION TO THE LAPW METHOD,\" in Planewaves, pseudopotentials, and the LAPW method, 2nd ed. New York, NY: Springer, 2006, pp. 43-52.
+    """
+
+    type = Quantity(
+        type=MEnum('apw', 'lapw', 'slapw'),  # ? where to put 'spherical_dirac'
+        description="""
+        Type of augmentation contribution. Abbreviations stand for:
+        | name | description | radial product |
+        |------|-------------|----------------|
+        | APW  | augmented plane wave with a frozen energy parameter | $A_{lm, k_n} u_l (r, E_l)$ |
+        | LAPW | linearized augmented plane wave with an optimized energy parameter | $A_{lm, k_n} u_l (r, E_l) + B_{lm, k_n} \dot{u}_{lm} (r, E_l^')$ |
+        | SLAPW | super linearized augmented plane wave | -- |
+        | spherical Dirac | spherical Dirac basis set | -- |
+
+        * http://susi.theochem.tuwien.ac.at/lapw/
+        """,
+    )
+
+    def n_terms_to_type(self, n_terms: Optional[int]) -> Optional[str]:
+        """
+        Set the type of the APW orbital based on the differential order.
+        """
+        if n_terms is None or n_terms == 0:
+            return None
+        if n_terms == 1:
+            return 'apw'
+        elif n_terms == 2:
+            return 'lapw'
+        else:
+            return 'slapw'
+
+    @check_normalized
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super().normalize(archive, logger)
+        # assign a APW orbital type
+        # this snippet works of the previous normalization
+        new_type = self.n_terms_to_type(self.n_terms)
+        if self.type is None:
+            self.type = new_type
+        elif self.type != new_type:
+            logger.error(
+                f'Inconsistent `APWOrbital` type: {self.type}. Setting back to `None`.'
+            )
+            self.type = None
```

Comment:
so I am not sure deleting info from the schema is a good idea

I think a logger.error is more than enough to understand this situations, so please, can you delete this assignment and the one in `APWBaseOrbital`?

Reply:
> so I am not sure deleting info from the schema is a good idea

Yeah, this is a general policy choice that we ought to discuss in Area C, heck even with the other Areas.

Here's my reasoning for setting it back: the logs are not hard-tied to the `archive.json`, i.e. we cannot guarantee that a user will see them together. So a user may naively assume the data to be correct, while a check will reveal an inconsistency. I'd rather have them skip this entry than consume incorrect data.

This is also related to the storage of the original data:
- for parsers, the user can always go to the original upload. Inconsistencies are more so a sign of bad parser logic.
- for ELNs, the user inputs the data, so overwriting it there may lead to data loss and is not advisable.